### PR TITLE
Onboarding prow build cluster: using gsm to stop sharing plain text secret

### PR DESCRIPTION
**Pull request by chaodaiG**
Also make the script smarter by helping creating PR

Note: previous implementation was done by kustomize, which unfortunately was not able to preserve boilerplate or inline comments, thus not suitable for k8s prow, since both exist, so fallback to use less strict `sed` replacing, luckily it works, at least for now..

### CODE REVIEW

**Review by fejta — APPROVED** [review]
/hold

**Comment by matthyx** [comment]
OMG, hate `sed`...
Do you need a LGTM?
/lgtm
/hold

**Comment by matthyx** [comment]
> /hold

oops, didn't refresh, sorry for the duplicate

File: prow/create-build-cluster.sh
Comment (cjwagner) [inline prow/create-build-cluster.sh:219]: This is a local variable of the caller so this won't be set, right?

Comment (chaodaiG) [reply prow/create-build-cluster.sh:219]: good catch! Didn't know how it worked in my local test, let me fix it

Comment (chaodaiG) [reply prow/create-build-cluster.sh:219]: https://github.com/kubernetes/test-infra/pull/21770

**Pull request merged**
